pkg/platform/packet/packet.go

```diff
@@ -248,3 +277,151 @@ func (c *config) GetExpectedNodes() int {
 
 	return c.ControllerCount + workers
 }
+
+// Check cluster config is valid
+// TODO: this function really needs to be tested
+func (c *config) checkValidConfig() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	diagnostics = append(diagnostics, c.checkNotEmptyWorkers()...)
+	diagnostics = append(diagnostics, c.checkWorkerPoolNamesUnique()...)
+	diagnostics = append(diagnostics, c.checkReservationIDs()...)
+
+	return diagnostics
+}
+
+func (c *config) checkNotEmptyWorkers() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	if len(c.WorkerPools) == 0 {
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "At least one worker pool must be defined",
+			Detail:   "Make sure to define at least one worker pool block in your cluster block",
+		})
+	}
+
+	return diagnostics
+}
+
+func (c *config) checkWorkerPoolNamesUnique() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	dup := make(map[string]bool)
+	for _, w := range c.WorkerPools {
+		if !dup[w.Name] {
+			dup[w.Name] = true
+			continue
+		}
+
+		// It is duplicated
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Worker pools name should be unique",
+			Detail:   fmt.Sprintf("Worker pool %v is duplicated", w.Name),
+		})
+	}
+
+	return diagnostics
+}
+
+func (c *config) checkReservationIDs() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	d := checkEachReservation(c.ReservationIDs, c.ReservationIDsDefault, "controller", c.ClusterName, c.ControllerCount)
+	diagnostics = append(diagnostics, d...)
+
+	for _, w := range c.WorkerPools {
+		d := checkEachReservation(w.ReservationIDs, w.ReservationIDsDefault, "worker", w.Name, w.Count)
+		diagnostics = append(diagnostics, d...)
+	}
+
+	return diagnostics
+
+}
+
+func checkEachReservation(reservationIDs map[string]string, resDefault, nodeRole, name string, count int) hcl.Diagnostics {
+	// This functions checks that hardware reservations are in the correct
```

Comment:
Function documentation should be above the function declaration.

Reply:
I'm not sure I want that to be documentation (is not exported function also), but I can move it if you prefer to have it as doc

Reply:
This makes it consistent with other function documentation I think.

Reply:
Moved it, thanks!